### PR TITLE
Don't re-hash the same objects

### PR DESCRIPTION
**Pull request by mariano54**

### CODE REVIEW

File: chia/wallet/puzzles/cat_loader.py [hunk 1]
```diff
@@ -2,3 +2,5 @@
 
 CAT_MOD = load_clvm("cat.clvm", package_or_requirement=__name__)
 LOCK_INNER_PUZZLE = load_clvm("lock.inner.puzzle.clvm", package_or_requirement=__name__)
+
+CAT_MOD_HASH = CAT_MOD.get_tree_hash()
```
Comment (arvidn) [inline chia/wallet/puzzles/cat_loader.py:6]: you don't seem to be using this anywhere. is that deliberate?

File: chia/wallet/cat_wallet/cat_utils.py [hunk 2]
```diff
@@ -41,11 +41,15 @@ def match_cat_puzzle(puzzle: Program) -> Tuple[bool, Iterator[Program]]:
         return False, iter(())
 
 
-def construct_cat_puzzle(mod_code: Program, limitations_program_hash: bytes32, inner_puzzle: Program) -> Program:
+def construct_cat_puzzle(
```
Comment (arvidn) [inline chia/wallet/cat_wallet/cat_utils.py:44]: aren't there any call sites to this function that needs updating too?

Comment (mariano54) [reply chia/wallet/cat_wallet/cat_utils.py:44]: yes but they are not so performance critical so i chose to make the patch smallerinstead

File: chia/wallet/puzzles/cat_loader.py (hunk 1 above)
Comment (mariano54) [reply chia/wallet/puzzles/cat_loader.py:6]: oh that was a bad cherry pick

Comment (mariano54) [reply chia/wallet/puzzles/cat_loader.py:6]: Please re-review

**Review by arvidn — DISMISSED** [review]

**Review by wjblanke — DISMISSED** [review]
aok

**Review by arvidn — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**